js/checkout.js

```diff
@@ -180,6 +203,67 @@ class Blockonomics {
     redirect_to_finish_order() {
         window.location.href = this.data.finish_order_url
     }
+
+    _set_refresh_loading(loading=false) {
+        if(loading) {
+            this._refresh.querySelector('.blockonomics-icon-refresh').classList.add('spin')
+            this._refresh.setAttribute('disabled', 'disabled')
+        } else {
+            this._refresh.querySelector('.blockonomics-icon-refresh').classList.remove('spin')
+            this._refresh.removeAttribute('disabled')
+        }
+    }
+
+    toggle_ajax_error(show=false) {
+        this._ajax_error.style.display = show ? 'block' : 'none'
+    }
+
+    refresh_order() {
+
+        this._set_refresh_loading(true)
+        this.toggle_ajax_error(false)
+
+        // Stop Progress Counter
+        clearInterval(this.progress.interval)
+        
+        fetch(this.data.get_order_amount_url, {method: 'GET'}).then(
+            res => {
+                // Enable the button anyways so that user can retry
+                this._set_refresh_loading(false)
+
+                if (!res.ok) {
+                    // Show Error, do not log to console as it is a Server Issue like 404
+                    this.toggle_ajax_error(true)
```

Comment:
I think whole toggle_ajax_error() is adding more complexity. Just reload the page if we are having ajax error

Reply:
That was the previous behavior, but Darren suggested that we show the errors to the User as it helps in debugging from the site owner's perspective and to see what's wrong or if anything is amiss. If we reload, it'll never be caught or reported.

Reply:
Understood. I don't expect lots of errors as everybody supports ajax. Just reloading is easy